narwhal/types/src/primary.rs

```diff
@@ -154,9 +155,72 @@ impl Hash<{ crypto::DIGEST_LENGTH }> for Batch {
     }
 }
 
+#[derive(Clone, Deserialize, MallocSizeOf, Serialize)]
+#[enum_dispatch(HeaderAPI)]
+pub enum Header {
+    V1(HeaderV1),
+}
+
+// TODO(arun): Is this default useful?
```

Comment:
One more question here @mystenmark. There is no way to get the committee info in the default trait so is just providing the default Header version you want to use the right way to handle it? Or can we version match somehow here 

cc @mwtian

Reply:
Do we actually use Header::default() outside of `Certificate::genesis()`? Making sure Header::default() is correct seems error prone.

Reply:
Nope thats the only place and I agree I would like to remove this impl but also dont know if something like Header::V1(HeaderV1::default()) in Certificate is any better

Reply:
I think we can iterate on the ergonomics later. Merging the layout change is more important for now.